sdk/eventgrid/azure-eventgrid/tests/test_eg_publisher_client.py

```diff
@@ -113,6 +114,51 @@ def test_send_cloud_event_data_dict(self, resource_group, eventgrid_topic, event
                 )
         client.send(cloud_event)
 
+    @CachedResourceGroupPreparer(name_prefix='eventgridtest')
+    @CachedEventGridTopicPreparer(name_prefix='cloudeventgridtest')
+    def test_send_cloud_event_data_base64_using_data(self, resource_group, eventgrid_topic, eventgrid_topic_primary_key, eventgrid_topic_endpoint):
+        akc_credential = AzureKeyCredential(eventgrid_topic_primary_key)
+        client = EventGridPublisherClient(eventgrid_topic_endpoint, akc_credential)
+        cloud_event = CloudEvent(
+                source = "http://samplesource.dev",
+                data = b'cloudevent',
+                type="Sample.Cloud.Event"
+                )
+
+        def callback(request):
+            req = json.loads(request.http_request.body)
+            assert req[0].get("data_base64") is not None
+            assert req[0].get("data") is None
```

Comment:
now I'm getting a bit confused by this request body.. because you set the `data` in the `CloudEvent` while the http request body `get("data")` is None,

so if I pass bytes to the `data` argument, which will internally set the `data_base64` variable?
```
cloud_event = CloudEvent(
    data = b'cloudevent',
)
# which of the following will have the value set?
cloud_event.data 
cloud_event.data_base64
```

or it's just implementation detail that users don't need to care about.

Reply:
internally, `cloud_event.data_base64` is set when we we pass bytes regardless of what we use in `data`, `data_base64` to send bytes.

Reply:
Based on the above, it might be useful to add a sample where you deserialize the CloudEvent that has the `data` field set to a binary value. Then, after deserializing, call `cloud_event.data_base64` and comment that it returns a not None value. Like [this sample](https://github.com/rakshith91/azure-sdk-for-python/blob/arch_preview_feedback/sdk/eventgrid/azure-eventgrid/samples/champion_scenarios/sample_consume_custom_payload.py), except with the data field set to `b'cloudevent'` like the test above.

Reply:
we do not have a `from_dict` implementation yet - it'd make more sense to add the samples after we have a symmetric workflow (plus i would like to unblock the breaking changes first)